version: '3.9'

services:
  proxy:
    container_name: apollo_proxy
    image: traefik:v2.6
    command:
      - "--log.level=DEBUG"
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--providers.docker.exposedbydefault=false"
      - "--entrypoints.http.address=:80"
      - "--entrypoints.https.address=:443"
    restart: always
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./config/static.yml:/etc/traefik/traefik.yml:ro
      - ./config/dynamic.yml:/etc/traefik/dynamic.yml:ro
      - ./certs:/etc/certs:ro
      - ./logs/traefik:/var/log
    networks:
      - apollo-external
      - apollo-internal
    
  gateway:
    container_name: apollo_gateway
    image: quay.io/datawire/ambassador:0.86.1
    command: --port 8080
    volumes:
      - ./config/gateway:/ambassador/ambassador-config
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.gateway.rule=Host(`api.${HOST}`)"
      - "traefik.http.routers.gateway.service=gateway"
      - "traefik.http.services.gateway.loadbalancer.server.port=8080"
      - "traefik.docker.network=apollo-external"
    networks:
      - apollo-internal

networks:
  # docker network create apollo-external
  apollo-external:
    external: true
  apollo-internal:
    external: false
